torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1260,6 +1260,28 @@ def reference_sigmoid(x):
             return (1 / (1 + np.exp(-x)))
         return scipy.special.expit(x)
 
+    def reference_lgamma(x):
+        # scipy.special.gammaln returns `-inf` when input is `-inf`.
+        # While Pytorch, C and C++, all return `inf` when input is `-inf`.
+        # Reference:
+        # https://en.cppreference.com/w/cpp/numeric/math/lgamma
+        # https://en.cppreference.com/w/c/numeric/math/lgamma
+
+        # To handle the above discrepancy,
+        # we never pass `-inf` to scipy.special.gammaln.
```

Comment:
This workaround is more clever than this comment suggests. It's not only that this never passes -inf to scipy.special.gammaln, but that it replaces -inf with inf so the mapping -inf->inf occurs as expected.

Reply:
Should we phrase it as,
```
# To handle the above discrepancy,
# we never pass `-inf`  to scipy.special.gammaln and replace them with `inf`.
```

Reply:
That's pretty good, but it requires the reader make the connection about how -inf is mapped to inf.

"this replaces -inf with inf so values that were originally -inf map to inf as expected"?

Reply:
I think it is very verbose but atleast clear. Will use that thanks!